src/LEGO.AsyncAPI/Writers/SpecialCharacterStringExtensions.cs

```diff
@@ -183,11 +191,25 @@ internal static string GetYamlCompatibleString(this string input)
                 return $"'{input}'";
             }
 
-            // If string can be mistaken as a number, a boolean, or a timestamp,
-            // wrap it in quote to indicate that this is indeed a string, not a number, a boolean, or a timestamp
-            if (decimal.TryParse(input, NumberStyles.Float, CultureInfo.InvariantCulture, out var _) ||
-                bool.TryParse(input, out var _) ||
-                DateTime.TryParse(input, out var _))
+            // Handle lexemes that can be intperated as as string
+            // https://yaml.org/spec/1.2-old/spec.html#id2761292 
```

Comment:
```suggestion
            // https://yaml.org/spec/1.2.2/#10213-integer
```
Lets reference the newest edition which it looks like YamlDotNet adheres to.